elektronLogging/clsfnTaskDistOverheadLogger.go

```diff
@@ -0,0 +1,49 @@
+package elektronLogging
+
+import (
+	"os"
+	log "github.com/sirupsen/logrus"
+)
+
+type ClsfnTaskDistOverheadLogger struct {
+	LoggerImpl
+}
+
+func NewClsfnTaskDistOverheadLogger(logType int, prefix string) *ClsfnTaskDistOverheadLogger {
+	cLog := new(ClsfnTaskDistOverheadLogger)
+	cLog.Type = logType
+	cLog.SetLogFile(prefix)
+	return cLog
+}
+
+func (cLog *ClsfnTaskDistOverheadLogger) Log(logType int, level log.Level, logData log.Fields,message string) {
+	if cLog.Type == logType {
+
+		logger.SetLevel(level)
+        
+        if cLog.AllowOnConsole {
+            logger.SetOutput(os.Stdout)
+		    logger.WithFields(logData).Println(message)
+        }
+		
+		logger.SetOutput(cLog.LogFileName)
+		logger.WithFields(logData).Println(message)
+	}
+	if cLog.next != nil {
+		cLog.next.Log(logType, level, logData, message)
+	}
+}
+
+func (cLog *ClsfnTaskDistOverheadLogger) SetLogFile(prefix string) {
+
+	tskDistLogPrefix := prefix + config.TaskDistConfig.FilenameExtension
+	if logDir != "" {
```

Comment:
Where is log dir coming from? Shouldn't be a global

Reply:
It is defined in `createLogDir.go`. Since all these files belong to same package, I accessed it in this way.

Reply:
I think it will be better if it is encapsulated.

Reply:
Done

Reply:
looks good now thanks!